polyglot/piranha/src/cleanup_rules/go/rules.toml

```diff
@@ -0,0 +1,547 @@
+            [
+                (false)
+                (parenthesized_expression (false))
+                (true)
+                (parenthesized_expression (true))
+                (nil)
+                (parenthesized_expression (nil))
+            ]
+        ) @lhs
+        operator: "=="
+        right: (
+            [
+                (false)
+                (parenthesized_expression (false))
+                (true)
+                (parenthesized_expression (true))
+                (nil)
+                (parenthesized_expression (nil))
+            ]
+        ) @rhs
+    ) @binary_expression
+    (#eq? @lhs @rhs)
+)
+"""
+replace = "true"
+replace_node = "binary_expression"
+
+# Simplifies not equal identity comparison
+# Note that `nil != nil` is not compilable in Go, but compiles in tree-sitter
+#   true != true   -> false
+#   false != false -> false
+#   nil != nil     -> false
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_identity_not_equal"
+query = """
+(
+    (binary_expression
+        left: (
+            [
+                (false)
+                (parenthesized_expression (false))
+                (true)
+                (parenthesized_expression (true))
+                (nil)
+                (parenthesized_expression (nil))
+            ]
+        ) @lhs
+        operator: "!="
+        right: (
+            [
+                (false)
+                (parenthesized_expression (false))
+                (true)
+                (parenthesized_expression (true))
+                (nil)
+                (parenthesized_expression (nil))
+            ]
+        ) @rhs
+    ) @binary_expression
+    (#eq? @lhs @rhs)
+)
+"""
+replace = "false"
+replace_node = "binary_expression"
+
+# Dummy rule that acts as a junction for all statement based cleanups
+[[rules]]
+name = "statement_cleanup"
+
+# Before :
+#  if (true) { doSomething(); }
+# After :
+#  { doSomething(); }
+#
+# Before :
+#  if (true) { doSomething(); } else { doSomethingElse();}
+# After :
+#  { doSomething(); }
+#
+[[rules]]
+groups = ["if_cleanup"]
+name = "simplify_if_statement_true"
+query = """
+(
+    (if_statement
+        condition : (
+            [
+                (true)
+                (parenthesized_expression (true))
+            ]
+        )
+        consequence : ((block) @consequence)
+    ) @if_statement
+)
+"""
+replace = "@consequence"
+replace_node = "if_statement"
+
+# Before :
+#  if (true) { doSomething(); } else { doSomethingElse();}
+# After :
+#  { doSomethingElse(); }
+#
+# Before :
+#  if (true) { doSomething(); }
+# After :
+#
+[[rules]]
+groups = ["if_cleanup"]
+name = "simplify_if_statement_false"
+query = """
+(
+    (if_statement
+        condition : (
+            [
+                (false)
+                (parenthesized_expression (false))
+            ]
+        )
+        consequence : (_)
+        alternative: ((_) @alternative) ?
+    ) @if_statement
+)
+"""
+replace = "@alternative"
```

Comment:
What happens in the case when there is no alternative?

Reply:
In this case, `@alternative` will be empty and the replacing works as intended: remove the whole if_statement. I added a test to cover this scenario.
```go
// no alternative, should remove the whole `if_statement`
if exp.BoolValue("false") {
	fmt.Println("to be removed 2")
}
```
leads to just
```go
// no alternative, should remove the whole `if_statement`
```